src/en/comichubfree/src/eu/kanade/tachiyomi/extension/en/comichubfree/ComicHubFree.kt

```diff
@@ -0,0 +1,145 @@
+package eu.kanade.tachiyomi.extension.en.comichubfree
+
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservableSuccess
+import eu.kanade.tachiyomi.source.Source
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import okhttp3.Request
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+class ComicHubFree : Source, ParsedHttpSource() {
+    override val baseUrl = "https://comichubfree.com"
+
+    override val lang = "en"
+
+    override val name = "ComicHubFree"
+
+    override val supportsLatest = true
+
+    override val client = network.cloudflareClient
+
+    override fun popularMangaSelector() = ".movie-list-index > .cartoon-box"
+
+    override fun latestUpdatesSelector() = popularMangaSelector()
+
+    override fun searchMangaSelector() = popularMangaSelector()
+
+    override fun popularMangaNextPageSelector() = "ul.pagination > li:last-child"
+
+    override fun latestUpdatesNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun searchMangaNextPageSelector() = popularMangaNextPageSelector()
+
+    override fun chapterListSelector() = "div.episode-list > div > table > tbody > tr"
+
+    override fun popularMangaRequest(page: Int): Request {
+        return GET("$baseUrl/popular-comic?page=$page")
+    }
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        return GET("$baseUrl/new-comic?page=$page")
+    }
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        return GET("$baseUrl/search-comic?key=$query&page=$page")
```

Comment:
```suggestion
        val url = "$baseUrl/search-comic".toHttpUrl().newBuilder().apply {
            addQueryParameter("key", query)
            addQueryParameter("page", page.toString())
        }.build()
        
        return GET(url, headers)
```
to url encode the query